### DISCUSSION TITLE
Give access to custom parameters in hooks

### DISCUSSION THREAD

**Discussion by sapientorius** [directus/directus#13786] [Feature Requests]
What I try to do is, to modify the return object if a certain parameter is true.

For example: https://example.com/items/things <= gets me all things

But with the parameter "modify=true" I want to apply a certain logic in the read filter. Like this: https://example.com/items/things?modify=true

Unfortunately I can not access the parameter "modify" since it is not one of the expected parameters and it doesn'r occour in any object I have access to.

**Comment by br41nslug** [comment]
Hooks do not necessarily have to be triggered within the context of an API request. Data can also be changed by other extensions or dataflows thus not having any of this information. You probably want to look into making a custom endpoint for this logic there you have access to any request data. https://docs.directus.io/extensions/endpoints/

**Reply by sapientorius** [reply]
Hey br41nslug,

thank you. Yes I know and I already did also custom endpoints, but in this case it would clearly more convinient to put that into a hook. I don't want to lose all the API features that come out of the box by using a custom endpoint.

**Comment by pkro** [comment]
This would be super useful. I want to enrich the returned data with expensive operations only for e.g. doing a report and not for all requests.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
